src/bootstrap/src/core/download.rs

```diff
@@ -648,7 +695,7 @@ HELP: if trying to compile an old commit of rustc, disable `download-rustc` in c
 download-rustc = false
 ";
         }
-        self.download_file(&format!("{base_url}/{url}"), &tarball, help_on_error);
+        self.download_file(&format!("{base_url}/{url}"), &tarball, help_on_error, &key.to_string());
```

Comment:
Please explicitly thread the commit through or rename the argument to `commit`. Otherwise changing the cache key will break this functionality in a confusing way.

You also shouldn't need to call .to_string() here.